src/hooks/useSyncChainQuery.ts

```diff
@@ -21,8 +21,18 @@ function getParsedChainId(parsedQs?: ParsedQs) {
 }
 
 export default function useSyncChainQuery() {
-  const { chainId, isActive } = useWeb3React()
+  const { chainId, isActive, account } = useWeb3React()
   const parsedQs = useParsedQueryString()
+  const chainIdRef = useRef(chainId)
+  const accountRef = useRef(account)
```

Comment:
Why is `accountRef` necessary? Why can't you just use `account` as a dep?

Reply:
Because account is originally `undefined` when pulled in from useWeb3React and later given a value if there is a connected wallet. We want to update the chainIdRef only when the account is first given that value so we need the context of if `account` has diverged from `accountRef`